sql/expression/arithmetic.go

```diff
@@ -223,12 +378,30 @@ func (a *Arithmetic) getArithmeticTypeFromExpr() sql.Type {
 			resType = r
 		}
 	} else if resType == nil {
-		return sql.Float64
+		return a.floatOrDecimal(lTyp, rTyp, lval, rval)
 	}
 
 	return resType
 }
 
+// getPrecisionAndScale converts the value to string format and parses it to get the precision and scale.
+func getPrecisionAndScale(val interface{}) (uint8, uint8) {
+	var str string
+	switch v := val.(type) {
+	case decimal.Decimal:
+		str = v.StringFixed(v.Exponent() * -1)
```

Comment:
If we already have a `decimal` type, can we just directly return its precision and scale instead of converting to a string and then reparsing?

Reply:
`decimal.Decimal` only has `.Exponent()` function that gives us the scale number, but no function for precision number. https://pkg.go.dev/github.com/shopspring/decimal#Decimal.NumDigits this is added in `v1.3.0` of the package, but we have `v1.2.0` package. Should I update the version and use this function? Reading the description of the function, it mentioned it's very slow fro bigger decimal number, but I'm not sure how slow this function is compared to parsing it from string in my implementation.

Reply:
Ahhh... that makes sense. The [implementation](https://github.com/shopspring/decimal/blob/master/decimal.go#L815) of `NumDigits` looks like it just converts to a string and grabs the length anyway, so no need to change anything here. I was hoping decimal was tracking those same precision/scale attributes, but I see we don't have that exact same interface. 

Upgrading to v1.3.0 looks like a good thing to do, but since it won't really make a difference here, no need to pull it in scope for this PR. Next time we look at upgrading dependencies, we should pick up that version bump though.